x/poe/contract/tgrade_validator_voting_integration_test.go

```diff
@@ -74,6 +86,64 @@ func TestValidatorsGovProposal(t *testing.T) {
 				assert.Equal(t, exp, gotPlan)
 			},
 		},
+		"cancel upgrade": {
+			src: contract.ValidatorProposal{
+				CancelUpgrade: &struct{}{},
+			},
+			assertExp: func(t *testing.T, ctx sdk.Context) {
+				_, exists := example.UpgradeKeeper.GetUpgradePlan(ctx)
+				assert.False(t, exists, "does not exist")
+			},
+		},
+		"update block params": {
+			src: contract.ValidatorProposal{
+				UpdateConsensusBlockParams: &contract.ConsensusBlockParamsUpdate{
+					MaxBytes: 10000000,
+					MaxGas:   20000000,
+				},
+			},
+			assertExp: func(t *testing.T, ctx sdk.Context) {
+				consensusParams := example.BaseApp.GetConsensusParams(ctx)
+				var expConsensusParams = app.DefaultConsensusParams
+				expConsensusParams.Block = &abci.BlockParams{
+					MaxBytes: 10000000,
+					MaxGas:   20000000,
+				}
+				assert.Equal(t, expConsensusParams, consensusParams)
+			},
+		},
+		"update evidence params": {
+			src: contract.ValidatorProposal{
+				UpdateConsensusEvidenceParams: &contract.ConsensusEvidenceParamsUpdate{
+					MaxAgeNumBlocks: 1000000,
+					MaxAgeDuration:  2000000,
+					MaxBytes:        3000000,
+				},
+			},
+			assertExp: func(t *testing.T, ctx sdk.Context) {
+				consensusParams := example.BaseApp.GetConsensusParams(ctx)
+				var expConsensusParams = app.DefaultConsensusParams
+				expConsensusParams.Evidence = &tmproto.EvidenceParams{
+					MaxAgeNumBlocks: 1000000,
+					MaxAgeDuration:  2000000 * time.Second,
+					MaxBytes:        3000000,
+				}
+				assert.Equal(t, expConsensusParams, consensusParams)
+			},
+		},
+		"migrate": {
+			src: contract.ValidatorProposal{
+				MigrateContract: &contract.Migration{
+					Contract:   valVotingAddr.String(),
+					CodeId:     codeID,
+					MigrateMsg: []byte("{}"),
+				},
+			},
+			assertExp: func(t *testing.T, ctx sdk.Context) {
+				gotContractInfo := example.TWasmKeeper.GetContractInfo(ctx, valVotingAddr)
+				assert.Equal(t, codeID, gotContractInfo.CodeID)
+			},
+		},
```

Comment:
L150: to ensure that tests do not affect others state they need to rollback (or not commit in our case). 
This can be done with 
```go
ctx, _ := parentCtx.WithBlockHeight(parentCtx.BlockHeight() + 1).CacheContext()
```
Where `parentCtx` is the context (state) that should be persisted for all in the setup

Reply:
Thanks for that as well. Will use it if needed.